server/server.go

```diff
@@ -772,6 +782,10 @@ func (s *server) Stop(ctx context.Context, req *rpcpb.StopRequest) (*rpcpb.StopR
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
+	// cancel possible concurrent still running start
```

Comment:
this is somewhat related, probably check for started cluster `if info == nil` shall be removed from Stop(), so Stop() can be called even if the first health check had not returned yet

Reply:
I am not sure I fully understood what you mean, and also, hope there is no other consequence, but I provided an update to this thinking to have addressed it.